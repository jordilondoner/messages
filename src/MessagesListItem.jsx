import React from 'react';
import moment from 'moment';

export default ({users, message}) => {
  const user = users.isPending ? false : users.users.find(i=> i.id === message.userId);
  return (
    <li className="message-item">
      <div className="avatar">
        {!user ? 'Loading user...' : <img alt={`${user.firstName}'s avatar`} src={user.avatar}/>}
      </div>
      <div className="message">
        {message.message}
        <span> On: {moment(new Date(message.timestamp)).format('LLLL')}</span>
        <span className="email">
          {!user ? 'Loading user...' : <p>{user.email}</p>}
        </span>
      </div>
    </li>
  )
}
