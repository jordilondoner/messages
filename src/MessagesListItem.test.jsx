import React from 'react';
import renderer from 'react-test-renderer';
import MessagesListItem from './MessagesListItem';
const messages = require('./messages.json');
const users = require('./members.json');

it('renders correctly', () => {
  const tree = renderer.create(
    <MessagesListItem
      message={messages[0]}
      users={{isPending: false, users}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
