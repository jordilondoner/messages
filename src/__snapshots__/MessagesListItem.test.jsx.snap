exports[`test renders correctly 1`] = `
<li
  className="message-item">
  <div
    className="avatar">
    <img
      alt="Albert\'s avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
  </div>
  <div
    className="message">
    Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.
    <span>
       On: 
      Thursday, February 9, 2017 4:27 AM
    </span>
    <span
      className="email">
      <p>
        arosec@bbb.org
      </p>
    </span>
  </div>
</li>
`;
